hr_attendance_calendar/models/hr_attendance_day.py

```diff
@@ -139,43 +116,43 @@ def _compute_due_hours(self):
                     if leave.state not in ['validate', 'validate1']:
                         continue
                     if not leave.holiday_status_id.remove_from_due_hours:
-                        # todo: use config instead of 8.0
+                        working_day = float(self.env[
+                            'ir.config_parameter'].get_param(
+                            'hr_attendance_calendar.work_day_duration'))
                         due_hours -= leave.compute_work_day(
-                            att_day.date) * 8
+                            att_day.date) * working_day
 
             att_day.due_hours = due_hours
 
     @api.multi
     @api.depends('attendance_ids.worked_hours')
     def _compute_worked_hours(self):
         for att_day in self.filtered('attendance_ids'):
-            total_hours = 0
-            for attendance in att_day.attendance_ids:
-                total_hours += attendance.worked_hours
-            att_day.worked_hours = total_hours
+            att_day.logged_hours = sum(
+                att_day.attendance_ids.mapped('worked_hours'))
 
     @api.multi
-    @api.depends('worked_hours')
+    @api.depends('logged_hours')
```

Comment:
You cannot have a compute field that depends on itself, something is wrong here.